packages/sdk/src/tables/emps/table.ts

```diff
@@ -1,8 +1,10 @@
 import { Data, makeId } from "./utils";
-import { JsMap } from "../generic";
+import BaseTable from "../base";
+import type { Store } from "../../stores";
+import { JsMap } from "../../stores";
 
-export const Table = (type = "Emp") => {
-  const table = JsMap<string, Data>(type, makeId);
+export const Table = (type = "Emp", store: Store<string, Data> = JsMap()) => {
+  const table = BaseTable<string, Data, Store<string, Data>>({ type, makeId }, store);
 
   async function addSponsors(id: string, sponsors: string[]) {
     const data = await table.get(id);
```

Comment:
2. here is the **issue**: `get()` returns `Promise<Data | undefined>`, so the type is inherited from the store object spread [here](https://github.com/UMAprotocol/protocol/pull/3376/files#diff-e3d44be706c5c9db01b1a83dffaac767e06f1e3c74737c1fc6003d1c78ffea43R31), not the `get()` function that overwrites it
And this happens in multiple cases, this is just an example.

I think I can avoid this, but I want to understand why it happens. It's related to the use of Generics I think